spec/section_spec.rb

```diff
@@ -44,24 +63,71 @@ class Page < SitePrism::Page; end
     expect(SitePrism::Section).to respond_to(:elements)
   end
 
-  it 'passes a given block to Capybara.within' do
-    expect(Capybara).to receive(:within).with('div')
+  describe 'instance' do
+    it 'passes a given block to Capybara.within' do
+      expect(Capybara).to receive(:within).with(locator)
 
-    SitePrism::Section.new(a_page, 'div') { 1 + 1 }
-  end
+      section_with_block
```

Comment:
This seems like it's the subject in this test.  Maybe consider splitting this into contexts with/without blocks? Or just move that code from the let down here.